src/main/java/org/scijava/table/io/TableIOOptions.java

```diff
@@ -0,0 +1,290 @@
+/*
+ * #%L
+ * Table structures for SciJava.
+ * %%
+ * Copyright (C) 2012 - 2020 Board of Regents of the University of
+ * Wisconsin-Madison, and Friedrich Miescher Institute for Biomedical Research.
+ * %%
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright notice,
+ *    this list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright notice,
+ *    this list of conditions and the following disclaimer in the documentation
+ *    and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDERS OR CONTRIBUTORS BE
+ * LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ * #L%
+ */
+
+package org.scijava.table.io;
+
+import org.scijava.optional.AbstractOptions;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.function.Function;
+
+/**
+ * @author Deborah Schmidt
+ */
+public class TableIOOptions extends AbstractOptions<TableIOOptions> {
+
+	public final Values values = new Values();
+	private final Map<Integer, ColumnTableIOOptions> columnOptions;
+	private static final String readColHeadersId = "readColHeaders";
+	private static final String columnOptionsId = "columnOptions";
+	private static final String writeColHeadersId = "writeColHeaders";
+	private static final String readRowHeadersId = "readRowHeaders";
+	private static final String writeRowHeadersId = "writeRowHeaders";
+	private static final String columnDelimiterId = "columnDelimiter";
+	private static final String rowDelimiterId = "rowDelimiter";
+	private static final String quoteId = "quote";
+	private static final String cornerTextId = "cornerText";
+	private static final String guessParserId = "guessParser";
+	private static final String parserId = "parser";
+	private static final String formatterId = "formatter";
```

Comment:
Instead of `<option>Id`, I'd suggest `<option>Key` to match the [`setValue(key,value)`](https://github.com/scijava/scijava-optional/blob/abb236e79bcf4b7668150daa486e809332a27a66/src/main/java/org/scijava/optional/Options.java#L5) signature in `Options`.